### DISCUSSION TITLE
Hi, created a new UI for ONNX, someone wants to give feedback of it?

### DISCUSSION THREAD

**Discussion by NeusZimmer** [Amblyopius/Stable-Diffusion-ONNX-FP16#26] [Show and tell]
Hi,
I just published a new UI version within my github profile, if someone want to have a look and try, will be good to have some feedback.
A couple of months ago I just landed here and started to learn about ONNX and Stable Diffussion, 
Then, I just were a little surprised that some options available for Automatic1111 were not implemented on the UI's I found, I just tried to modified them a little bit and ended in something little bit more complex.

Current version: 
Available pipelines: txt2img, img2img, Inpaint, instruct pix2pix.

Allows modification of the  pipeline providers without re-running the UI., also, you may want to run some pipeline in one graphic card, another card for VAE and CPU to the rest... this UI allows such granularity for :main model, schedulers, VAE, text encoder...

Allows the use of a different VAE for a model ( I tested and many models got the same VAE , then, why keep storing them on disk?

Add a clean memory option: do  not know why, but changing resolution sometimes keep garbage in memory, and that ends making an impact on the time needed for the inferences.

Also, wildcards, a working deepdanbooru interrogator and resolution increase option. (one of first tests with a ONNX model, a MS Model ([super-resolution-10.onnx]) to increase file resolution up to crazy sizes by slicing & joining (working , but not for professional uses)
PD: first python code I did... sure it could be improved, but working fine and easy to modify in future releases.

**Comment by Amblyopius** [comment]
Hi, I'll definitely have a look at it soon.

If you want to save disk space, a different lay out on disk is definitely a good idea. Essentially duplication exists in:
- Tokenizer -> Always the same (but tiny)
- text_encoder -> There's 2 common frozen CLIP text encoders
- VAE -> There's a few of them but generally it's better to reduce anyway as some of the older ones are better of being replaced
- Controlnet -> Storing them inside the model directory allows for autoloading in diffusers but it's a massive waste of diskspace as they don't change depending on what you match them with

The reload on resolution change is most likely a problem in the ONNX Runtime optimiser. While running it adapts to the resolution and it seems to have an issue if you use the same model again with a different resolution. Disadvantage of a reload is that the entire optimiser runs again, not just the part you need. It's possible to save targeted models, I need to check if that makes it feasible to swap fast so that a reload takes hardly any time.

**Reply by NeusZimmer** [reply]
Thanks for your reply, that's right, all those duplicated models might be a little more static in disk. I was pointing to the Controlnet model too, re-using, as it could help a lot the process, and you may not need to re-convert every model to work with Controlnet... but still in test stages... as I was focused on the re-using  of generated latents instead on finishing it... and will check on the new version of diffusers next 
EDIT: Checked, it works!!.  Looks like I was working on the torch nightly environment (I forgot that I did a clean install some days ago and it was working into it)

**Reply by NeusZimmer** [reply]
Now I adapted it to re-using the same ControlNet models, this just save good space on our disks... will make it work that way with TextEncoder soon.

**Comment by Monkey767666** [comment]
How can I use that WebUI?
Can I install a WebUI from another model?
How to do that?

**Reply by NeusZimmer** [reply]
I modified the install instructions, have a look at it.
